packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx

```diff
@@ -99,9 +124,108 @@ const CompactInteractiveList = (props) => {
     width,
     columnMinimumWidth,
     columnMaximumWidth,
+    onCellSelect,
+    onClearSelection,
   } = props;
 
   const theme = useContext(ThemeContext);
+  const listRef = React.useRef();
+
+  // using ref so that keyboard navigation and focusing on cells won't trigger component re-render.
+  const focusedCell = useRef({ row: 0, cell: 0 });
+
+  const setFocusOnCell = ({ row, cell }) => {
+    // add 1 to the row number to accomodate for hidden header
+    const focusedCellElement = listRef.current.children[row + 1].children[cell];
+    const interactiveChildren = getFocusableElements(focusedCellElement);
+    if (interactiveChildren?.length > 0 && interactiveChildren[0]) {
+      // currently a cell can have only one interact-able element in it, which gets auto-focused.
+      interactiveChildren[0].focus();
+    } else {
+      // cell gets focus if there is no interact-able elements in it.
+      focusedCellElement.focus();
+    }
+  };
+
+  const setFocusedRowCol = ({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    setFocusOnCell({ row, cell });
+  };
+
+  const handleOnCellSelect = useCallback(({ row, cell }) => {
+    focusedCell.current = { row, cell };
+    // check for interactive elements in that cell
+    // add 1 to the row number to accomodate for hidden header
+    const elementToFocus = listRef.current.children[row + 1].children[cell];
```

Comment:
If you are selecting the cell, it should already be getting focus.

Reply:
You are right, that cell already has focus. The logic behind this line has different purpose and I changed the naming from `elementToFocus` to be `focusedElement`.
This line is here as a stepping stone for detecting if currently focused cell has an interactive element, as per my understanding if the cell has an interactive element, the cell should not be selectable, hence onCellSelect method should not be called. 

@eawww, I wanted to double-check with you the logic in case the cell has interactive element in it. Does it make the cell non selectable, or did I get it wrong?